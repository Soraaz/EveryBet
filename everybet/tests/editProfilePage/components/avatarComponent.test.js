import React from "react";
import renderer from "react-test-renderer";

import AvatarComponent from "../../../src/editProfilePage/components/avatarComponent.js";

describe("AvatarComponent", () => {
	let onAvatarChange = jest.fn();

	describe("rendering", () => {
		it("Should render correctly", () => {
			const tree = renderer
				.create(<AvatarComponent onAvatarChange={onAvatarChange}/>)
				.toJSON();
			expect(tree).to.matchSnapshot();
		});
	});

	describe("functionalities", () => {
		let component;

		beforeEach(() => {
			component = new AvatarComponent();
		});

		it("Should instantiate the class", () => {
			expect(component).to.be.an.instanceOf(AvatarComponent);
		});

		/*		it("Should be testing askPermissions", (done) => {
					component.askPermission();
						expect(component).to.be.an.instanceOf(AvatarComponent);
				});*/
	});
});
