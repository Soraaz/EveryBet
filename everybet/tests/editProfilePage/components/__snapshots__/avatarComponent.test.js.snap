// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarComponent rendering Should render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "center",
      "marginBottom": 30,
      "marginTop": 30,
    }
  }
>
  <View
    style={
      Object {
        "marginLeft": 30,
        "marginRight": 30,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../assets/images/user_placeholder.png",
        }
      }
      style={
        Object {
          "alignSelf": "center",
          "borderRadius": 40,
          "height": 80,
          "width": 80,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
        "marginRight": 50,
      }
    }
  >
    <Text />
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "#FFFFFF",
          "borderRadius": 10,
          "height": 30,
          "justifyContent": "center",
          "marginTop": 20,
          "opacity": 1,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 6,
            "width": 0,
          },
          "shadowOpacity": 0.33,
          "shadowRadius": 7.5,
          "width": 171,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontSize": 15,
            "textAlign": "center",
          }
        }
      >
        Changer d'avatar
      </Text>
    </View>
  </View>
</View>
`;
